src/components/TradeWidget/index.tsx

```diff
@@ -438,19 +438,19 @@ const TradeWidget: React.FC = () => {
     // We DO want to use price estimation when there's no price coming from the URL
     const shouldUsePriceEstimation = !initialPrice.current || +initialPrice.current === 0
 
-    // Only bother when there's an actual priceEstimation
-    // This is important because on first page load, default price is `null`
-    // Only after the promise is resolved there will be anything
-    if (priceEstimation) {
-      // There's a valid price estimation, we can clear initial price.
-      // Keep in mind that next time is when token selection is changed.
-      // In that case, initial price doesn't matter anymore
-      initialPrice.current = ''
+    // Only try to update price estimation when not loading
+    if (!isPriceLoading) {
+      // If there was a price set coming from the URL, reset it
+      // It was supposed to be used only once. Initial price doesn't matter anymore
+      if (initialPrice.current) {
+        initialPrice.current = ''
+      }
 
-      logDebug('[TradeWidget] priceEstimation', priceEstimation.toString(10))
+      logDebug(`[TradeWidget] priceEstimation ${priceEstimation}`)
 
       if (shouldUsePriceEstimation) {
-        const newPrice = priceEstimation.toFixed(5)
+        // Price estimation can be null. In that case, set the input to 0
+        const newPrice = priceEstimation ? priceEstimation.toFixed(5) : '0'
```

Comment:
Why `5` ?

Reply:
Because reasons.

Seriously, how far do we want to go precision wise? Does it make sense to input a 18 decimals price?

Reply:
No, I think it's fine for the estimation, but maybe a const would be good. `5` look like a magic number. Just a nit

Reply:
Makes sense, done.